03_file_structure_analysis.R

```diff
@@ -228,111 +228,43 @@ export_id_client <- Client %>%
-    CulturallySpecific = CulturallySpecific %in% c(yes_no),
-    DifferentIdentity = DifferentIdentity %in% c(yes_no),
-    Questioning = Questioning %in% c(yes_no),
-    GenderNone = GenderNone %in% c(dkr_dnc) | is.na(GenderNone)
-  ) %>%
-  group_by_all() %>%
-  summarise(
-    VeteranStatus = min(VeteranStatus, na.rm = FALSE),
-    RaceNone = min(RaceNone, na.rm = FALSE),
-    AmIndAKNative = min(AmIndAKNative, na.rm = FALSE),
-    Asian = min(Asian, na.rm = FALSE),
-    BlackAfAmerican = min(BlackAfAmerican, na.rm = FALSE),
-    NativeHIPacific = min(NativeHIPacific, na.rm = FALSE),
-    White = min(White, na.rm = FALSE),
-    MidEastNAfrican = min(MidEastNAfrican, na.rm = FALSE),
-    HispanicLatinaeo = min(HispanicLatinaeo, na.rm = FALSE),
-    Woman = min(Woman, na.rm = FALSE),
-    Man = min(Man, na.rm = FALSE),
-    NonBinary = min(NonBinary, na.rm = FALSE),
-    Transgender = min(Transgender, na.rm = FALSE),
-    CulturallySpecific = min(CulturallySpecific, na.rm = FALSE),
-    DifferentIdentity = min(DifferentIdentity, na.rm = FALSE),
-    Questioning = min(Questioning, na.rm = FALSE),
-    GenderNone = min(GenderNone, na.rm = FALSE)
-  ) %>%
-  ungroup() %>%
-  select(
-    VeteranStatus,
-    RaceNone,
-    AmIndAKNative,
-    Asian,
-    BlackAfAmerican,
-    NativeHIPacific,
-    White,
-    MidEastNAfrican,
-    HispanicLatinaeo,
-    Woman,
-    Man,
-    NonBinary,
-    Transgender,
-    CulturallySpecific,
-    DifferentIdentity,
-    Questioning,
-    GenderNone
-  ) %>%
-  pivot_longer(cols = everything()) %>%
-  filter(value == 0) %>%
-  count(name) %>%
+# CHECK: Invalid demographic values
+# first, get a mapping of variables and their expected values
+cols <- c("VeteranStatus", "RaceNone", "AmIndAKNative", "Asian", "BlackAfAmerican", 
+          "NativeHIPacific", "White", "MidEastNAfrican", "HispanicLatinaeO", 
+          "Woman", "Man", "NonBinary", "Transgender", "CulturallySpecific",
+          "DifferentIdentity", "Questioning", "GenderNone")
+
+valid_values <- list(yes_no_enhanced, dkr_dnc, yes_no, yes_no, yes_no, yes_no, 
+                     yes_no, yes_no, yes_no, yes_no, yes_no, yes_no, yes_no,
+                     yes_no, yes_no, yes_no, dkr_dnc)
+
+
+# Only take existing columns - this solves the issue of misspelled demographic 
+# columns
+existing_cols <- intersect(cols, names(Client))
+
+# Create a named list of valid values for existing columns
+valid_values_named <- setNames(valid_values, cols)[existing_cols]
+
+# looping through only the columns that are actually (not misspelled) in Client
+# check if it's an unexpected, non-na value. That's what the [[.]] does -
+# it refers to the particular column in the loop. Equivalent to pull(.)
+# The ~ defines an anonymous function, as opposed to creating a specific function
+get_unexpected_count <- function(col_name) {
+  unexpected <- !Client[[col_name]] %in% valid_values_named[[col_name]] & 
+    !is.na(Client[[col_name]])
```

Comment:
is.na() is actually valid for RaceNone and GenderNone. So we probably want to revise to something like:

`unexpected <- !Client[[col_name]] %in% valid_values_named[[col_name]] | 
    (col_name %in% c("RaceNone","GenderNone") & is.na(Client[[col_name]]))`